### PR TITLE
feat: @W-11668812 explicitly import/export types

### PR DESCRIPTION
**Pull request by jdalton**
Explicitly import/export types. This helps transpilers and humans reading things.

### CODE REVIEW

File: packages/near-membrane-base/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "main": "dist/index.cjs.js",
     "module": "dist/index.js",
     "sideEffects": false,
-    "types": "types/index.d.ts",
```
Comment (jdalton) [inline packages/near-membrane-base/package.json:10]: consistent with LWS which uses `typings`

File: examples/.rolluprc.cjs [hunk 2]
```diff
@@ -8,11 +9,15 @@ function config(dir) {
...
+        plugins: [
+            nodeResolve({
+                preferBuiltins: true,
+            })
+        ],
     };
 }
 
-export default [
+module.exports = [
```
Comment (rwaldron) [inline examples/.rolluprc.cjs:20]: But why?

File: packages/near-membrane-base/src/environment.ts [hunk 3]
```diff
@@ -1,31 +1,18 @@
-import {
+import type {
```
Comment (rwaldron) [inline packages/near-membrane-base/src/environment.ts:1]: Excellent

File: packages/near-membrane-base/src/membrane.ts [hunk 4]
```diff
@@ -23,178 +23,57 @@
...
+    HooksOptions,
+    Pointer,
+    PointerOrPrimitive,
+    PropertyKey,
+    PropertyKeys,
+    ProxyTarget,
+    SerializedValue,
+    Setter,
+    ShadowTarget,
+} from './types';
```
Comment (rwaldron) [inline packages/near-membrane-base/src/membrane.ts:76]: Tidy AF.

File: packages/near-membrane-base/src/membrane.ts [hunk 5]
```diff
@@ -4748,27 +4629,33 @@ export function createMembraneMarshall(
...
-            BoundaryProxyHandlerProto[arityToApplyTrapNameRegistry[5]] = applyTrapForFiveOrMoreArgs;
-            BoundaryProxyHandlerProto[arityToApplyTrapNameRegistry.n] = applyTrapForAnyNumberOfArgs;
-            BoundaryProxyHandlerProto[arityToConstructTrapNameRegistry[0]] =
+            (BoundaryProxyHandlerProto as any)[arityToApplyTrapNameRegistry[4]] =
+                applyTrapForFourOrMoreArgs;
+            (BoundaryProxyHandlerProto as any)[arityToApplyTrapNameRegistry[5]] =
+                applyTrapForFiveOrMoreArgs;
+            (BoundaryProxyHandlerProto as any)[arityToApplyTrapNameRegistry.n] =
+                applyTrapForAnyNumberOfArgs;
+            (BoundaryProxyHandlerProto as any)[arityToConstructTrapNameRegistry[0]] =
```
Comment (rwaldron) [inline packages/near-membrane-base/src/membrane.ts:4646]: So minor, but definitely makes this easier to read.

File: packages/near-membrane-base/src/types.ts [hunk 6]
```diff
@@ -1,4 +1,194 @@
...
+export type ProxyTarget = CallableFunction | any[] | object;
+export type SerializedValue = bigint | boolean | number | string | symbol;
 export type Setter = (value: any) => void;
+export type ShadowTarget = ProxyTarget;
+export interface VirtualEnvironmentOptions {
+    blueConnector: Connector;
+    redConnector: Connector;
+    distortionCallback?: DistortionCallback;
+    instrumentation?: Instrumentation;
+}
```
Comment (rwaldron) [inline packages/near-membrane-base/src/types.ts:194]: I love this!!

(The whole file)

File: tsconfig.json [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
-        "module": "es2020",
-        "moduleResolution": "node",
         "esModuleInterop": true,
         "lib": ["dom", "es2015", "es2016", "es2017", "es2018", "es2019", "es2020"],
-        "declaration": false,
-        // Enhance Strictness
-        "strict": true,
-        "noUnusedLocals": true,
-        "suppressImplicitAnyIndexErrors": true,
+        "module": "commonjs",
```
Comment (rwaldron) [inline tsconfig.json:5]: Why is this changed?

**Review by rwaldron — APPROVED** [review]
Approved, with some clarifying questions

File: examples/.rolluprc.cjs (hunk 2 above)
Comment (jdalton) [reply examples/.rolluprc.cjs:20]: Consistency with Locker

File: tsconfig.json (hunk 7 above)
Comment (jdalton) [reply tsconfig.json:5]: Just more consistency with. Locker.
